samples/bluetooth/central_hids/src/main.c

```diff
@@ -247,6 +340,139 @@ static const struct bt_gatt_hids_c_init_params hids_c_init_params = {
 	.pm_update_cb  = hids_c_pm_update_cb
 };
 
+
+static void button_bootmode(void)
+{
+	if (!bt_gatt_hids_ready_check(&hids_c)) {
+		printk("HID device not ready\n");
+		return;
+	}
+	int err;
+	enum bt_gatt_hids_c_pm pm = bt_gatt_hids_c_pm_get(&hids_c);
+
+	printk("Setting protocol mode: %s\n",
+	       (pm == BT_GATT_HIDS_C_PM_BOOT) ?
+	       "BOOT" : "REPORT");
+	err = bt_gatt_hids_c_pm_write(&hids_c,
+				      (pm == BT_GATT_HIDS_C_PM_BOOT) ?
+				      BT_GATT_HIDS_C_PM_REPORT :
+				      BT_GATT_HIDS_C_PM_BOOT);
+	if (err) {
+		printk("Cannot change protocol mode (err %d)\n", err);
+	}
+}
+
+
+static void button_capslock(void)
+{
+	int err;
+	u8_t data;
+
+	if (!bt_gatt_hids_ready_check(&hids_c)) {
+		printk("HID device not ready\n");
+		return;
+	}
+	if (!hids_c.rep_boot.kbd_out) {
+		printk("HID device does not have Keyboard OUT report\n");
+		return;
+	}
+	if (bt_gatt_hids_c_pm_get(&hids_c) != BT_GATT_HIDS_C_PM_BOOT) {
+		printk("This function works only in BOOT Report mode\n");
+		return;
+	}
+	capslock_state = capslock_state ? 0 : 1;
+	data = capslock_state ? 0x02 : 0;
+	err = bt_gatt_hids_c_rep_write_wo_rsp(&hids_c,
+					      hids_c.rep_boot.kbd_out,
+					      &data, sizeof(data));
+	if (err) {
+		printk("Keyboard data write error (err: %d)\n", err);
+		return;
+	}
+	printk("Caps lock send (val: 0x%x)\n", (unsigned int)data);
```

Comment:
casting to `unsigned int` is not needed as default argument promotion will do it by itself

Reply:
also in other places

Reply:
done